packages/components/src/checkbox-control/index.tsx

```diff
@@ -15,17 +16,43 @@ import { Icon, check, reset } from '@wordpress/icons';
  * Internal dependencies
  */
 import BaseControl from '../base-control';
+import type { CheckboxControlProps } from './types';
+import type { WordPressComponentProps } from '../ui/context';
+
+/**
+ * Checkboxes allow the user to select one or more items from a set.
+ *
+ * ```jsx
+ * import { CheckboxControl } from '@wordpress/components';
+ * import { useState } from '@wordpress/element';
+ *
+ * const MyCheckboxControl = () => {
+ *   const [ isChecked, setChecked ] = useState( true );
+ *   return (
+ *     <CheckboxControl
+ *       label="Is author"
+ *       help="Is the user a author or not?"
+ *       checked={ isChecked }
+ *       onChange={ setChecked }
+ *     />
+ *   );
+ * };
+ * ```
+ */
+export function CheckboxControl(
+	props: WordPressComponentProps< CheckboxControlProps, 'input', false >
```

Comment:
```suggestion
	// ref is omitted until we have `WordPressComponentPropsWithoutRef` or add
	// ref forwarding to CheckboxControl.
	props: Omit<
		WordPressComponentProps< CheckboxControlProps, 'input', false >,
		'ref'
	>
```

This might be better done within `types.ts`?

Reply:
I belive the convention have been to use `WordPressComponentProps` in the compontent and not in `types.ts` so adding it here seems right.

Reply:
I don't feel strongly either way though there is a [precedent in the `InputControl`](https://github.com/WordPress/gutenberg/blob/trunk/packages/components/src/input-control/types.ts#L146-L163) for it being in `types.ts`.